### PR TITLE
[PAYOP-1292] Implement DCVV support

### PR DESCRIPTION
**Pull request by fangpenlin**
## Description of changes in release / Impact of release:
 Implement DCVV support

Also fix this https://verygoodsecurity.atlassian.net/browse/PAYOP-1291

## Documentation
(insert text here)

### Is this a breaking change?
- [ ] Yes
- [ ] No

### If you answered Yes then describe why is it so
(insert text here if applicable)

### Is there a way to disable the change?
- [ ] Use previous release
- [ ] Use a feature flag
- [ ] No

#### Additional details go here
(insert text here if applicable)

[PAYOP-1292]: https://verygoodsecurity.atlassian.net/browse/PAYOP-1292?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: larky/src/main/resources/vgs/nts.star [hunk 1]
```diff
@@ -34,7 +36,11 @@ def render(
     :param pan: JSONPath to the PAN alias in the input payload. Used to look up the corresponding network token to be
            rendered and injected into the payload.
     :param cvv: JSONPath to the CVV of the credit card in the input payload. Used to pass to the network for retrieving
-           the corresponding network token and cryptogram to be returned.
+           the corresponding network token and cryptogram to be returned. One of `cvv` or `dcvv` need to be provided,
+           but not both in the same time.
+    :param dcvv: JSONPath to the CVV of the credit card in the input payload. Used to pass to the network for retrieving
+           the corresponding network token and dynamic CVV to be returned. One of `cvv` or `dcvv` need to be provided,
+           but not both in the same time.
```
Comment (mjallday) [inline larky/src/main/resources/vgs/nts.star:43]:
```suggestion
           but not both at the same time.

```

File: larky/src/main/resources/vgs/nts.star [hunk 2]
```diff
@@ -49,15 +55,23 @@ def render(
...
     """
     pan_value = jsonpath_ng.parse(pan).find(input).value
-    cvv_value = jsonpath_ng.parse(cvv).find(input).value
     amount_value = str(jsonpath_ng.parse(amount).find(input).value)
     currency_code_value = jsonpath_ng.parse(currency_code).find(input).value
+    if cvv != None and dcvv == None:
+        cvv_value = jsonpath_ng.parse(cvv).find(input).value
+    elif dcvv != None and cvv == None:
+        cvv_value = jsonpath_ng.parse(dcvv).find(input).value
+    elif cvv == None and dcvv == None:
```
Comment (mjallday) [inline larky/src/main/resources/vgs/nts.star:64]:
```suggestion
    elif not any([cvv, dcvv]):
```

would capture empty string versus null. empty string is still not a valid value

File: larky/src/main/resources/vgs/nts.star [hunk 3]
```diff
@@ -49,15 +55,23 @@ def render(
     :return: JSON payload injected with network token values
     """
     pan_value = jsonpath_ng.parse(pan).find(input).value
-    cvv_value = jsonpath_ng.parse(cvv).find(input).value
     amount_value = str(jsonpath_ng.parse(amount).find(input).value)
     currency_code_value = jsonpath_ng.parse(currency_code).find(input).value
+    if cvv != None and dcvv == None:
```
Comment (mjallday) [inline larky/src/main/resources/vgs/nts.star:60]:
```python
if all([cvv, dcvv]):
  fail("ValueError: only either one of cvv or dvcc can be provided")
if not any([cvv, dcvv]):
  fail("ValueError: either one of cvv or dvcc need to be provided")
cvv_value = jsonpath_ng.parse(cvv or dvcc).find(input).value
```
may be simpler like this (just style, feel free to ignore)

**Review by mjallday — APPROVED** [review]
lgtm

**Review by viacheslav-fomin-main — APPROVED** [review]

**Pull request merged**